comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala

```diff
@@ -157,21 +155,17 @@ object TcpTransportLayer {
   type ConnectionsState = MonadState[Task, Connections]
 }
 
-class TransportLayerImpl(dispatch: ProtocolMessage => Task[CommunicationResponse])(
+class TransportLayerImpl(dispatch: Protocol => Task[CommunicationResponse])(
     implicit scheduler: Scheduler)
     extends TransportLayerGrpc.TransportLayer {
 
   def send(request: TLRequest): Future[TLResponse] =
     request.protocol
       .fold(internalServerError("protocol not available in request").pure[Task]) { protocol =>
-        ProtocolMessage.toProtocolMessage(protocol) match {
-          case Left(error) => internalServerError(error.toString).pure[Task]
-          case Right(pm) =>
-            dispatch(pm).map {
-              case NotHandled                   => internalServerError(s"Message $pm was not handled!")
-              case HandledWitoutMessage         => noResponse
-              case HandledWithMessage(response) => returnProtocol(response.proto)
-            }
+        dispatch(protocol) >>= {
+          case NotHandled                   => internalServerError(s"Message ${protocol} was not handled!").pure[Task]
+          case HandledWitoutMessage         => noResponse.pure[Task]
+          case HandledWithMessage(response) => returnProtocol(response).pure[Task]
```

Comment:
Please explain why we need a `flatMap` here. I changed it to `map` in one of my PRs because each case ends with a `.pure[Task]`.

Reply:
yes, map will suffice, good catch!